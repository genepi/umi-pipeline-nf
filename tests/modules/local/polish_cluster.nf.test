nextflow_process {

    name "Test Process POLISH_CLUSTER"
    script "modules/local/umi_polishing/polish_cluster.nf"
    process "POLISH_CLUSTER"

        setup{
            run("GLUE_CLUSTERS"){
                script "modules/local/umi_polishing/glue_clusters.nf"
                params{
                    min_reads_per_cluster = 10
                }
                process {
                    """
                    input[0] = Channel.of(
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule0_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule1_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule2_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule3_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule4_sub0.fastq"]
                    )
                    input[1] = "raw"
                    """
                }
            }

        }

    test("Should run without failures - one sample per file") {
        when {
            process {
                """
                    input[0] = Channel.of(
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule0_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule1_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule2_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule3_sub0.fastq"],
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/single_clusters/smolecule4_sub0.fastq"]
                    )
                    input[1] = "consensus"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.consensus_fastq).match()
        }

    }

    test("Should run without failures - multiple samples per file") {
        when {
            process {
                """
                    input[0] = Channel.of(
                        ["barcode03", "LPA2645", "$baseDir/tests/input/polish_cluster/glued_clusters/barcode03_combined_clusters_1.fastq"]
                    )
                    input[1] = "consensus"
                """
            }
        }

        then {
            assert process.success
            def fastqFile = path(process.out.consensus_fastq.get(0).get(2)).fastq
            assert fastqFile.getNumberOfRecords() == 6
        }

    }

    test("Should run without failures") {
        when {
            process {
                """
                input[0] = GLUE_CLUSTERS.out.glued_clusters
                            .map{ sample, type, clusters -> tuple(sample, type, clusters instanceof List ? clusters : [clusters]) }
                            .transpose(by: 2)
                input[1] = "consensus"
                """
            }
        }

        then {
            assert process.success 
            assert snapshot(process.out.consensus_fastq).match()
        }

    }

}
