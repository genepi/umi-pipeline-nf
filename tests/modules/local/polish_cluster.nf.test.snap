{
    "Should run without failures - one sample per file": {
        "content": [
            [
                [
                    "barcode03",
                    "LPA2645",
                    "smolecule0_sub0_consensus.fastq:md5,1e679a28156c5dde61438a93527a603c"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "smolecule1_sub0_consensus.fastq:md5,3d0413b0a65a319ff18f99abea419936"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "smolecule2_sub0_consensus.fastq:md5,ef1f8d62e65952b7918a4fac0f44873e"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "smolecule3_sub0_consensus.fastq:md5,45c378b6ea41b68c1e3fbbca806db1d3"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "smolecule4_sub0_consensus.fastq:md5,feef582d38b07e3d712abaa561fd0c54"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T11:58:44.875638"
    },
    "Should run without failures - multiple samples per file": {
        "content": [
            [
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,bb6c68801220b0e6c50026a5c826bee6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T11:59:03.581829"
    },
    "Should run without failures": {
        "content": [
            [
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,1e679a28156c5dde61438a93527a603c"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,3d0413b0a65a319ff18f99abea419936"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,45c378b6ea41b68c1e3fbbca806db1d3"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,ef1f8d62e65952b7918a4fac0f44873e"
                ],
                [
                    "barcode03",
                    "LPA2645",
                    "barcode03_combined_clusters_1_consensus.fastq:md5,feef582d38b07e3d712abaa561fd0c54"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T19:03:54.467047"
    }
}